include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -2251,6 +2258,27 @@ __parallel_partial_sort(oneapi::dpl::__internal::__device_backend_tag __backend_
                                         __partial_merge_kernel<decltype(__mid_idx)>{__mid_idx}, __comp);
 }
 
+//------------------------------------------------------------------------
+// parallel_partial_sort - async pattern
+//-----------------------------------------------------------------------
+
+// TODO: check if it makes sense to move these wrappers out of backend to a common place
+// TODO: consider changing __partial_merge_kernel to make it compatible with
+//       __full_merge_kernel in order to use __parallel_sort_impl routine
+template <typename _CustomName, typename _Iterator, typename _Compare>
+__future<sycl::event>
+__parallel_partial_sort(oneapi::dpl::__internal::__device_backend_tag __backend_tag, sycl::queue& __q,
+                        _Iterator __first, _Iterator __mid, _Iterator __last, _Compare __comp)
+{
+    const auto __mid_idx = __mid - __first;
+
+    auto __keep = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read_write, _Iterator>();
+    auto __buf = __keep(__first, __last);
+
+    return __parallel_partial_sort_impl<_CustomName>(__backend_tag, __q, __buf.all_view(),
+                                                     __partial_merge_kernel<decltype(__mid_idx)>{__mid_idx}, __comp);
+}
```

Comment:
It looks like this is meant to replace the previous function, but instead adds an overload.  I think we need to remove the old version.